src/timeout.c

```diff
@@ -175,7 +177,7 @@ int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout, int
             addReplyError(c, "timeout is out of range");
             return C_ERR;
         }
-        tval = (long long) ftval;
+        tval = (long long) ceill(ftval);
```

Comment:
One last ask is we should add a test for this. You can use something like https://github.com/redis/redis/blob/unstable/tests/unit/type/list.tcl#L1276 for reference. I am imagining a simple test like:
```
Create a deferring client.
Have it block with 0.001. 
Wait for the number of blocked clients to equal 0
Validate the empty read from the deferring client.
```

This way, even if it fails the test won't actually hang since the wait for the number of blocked clients to equal 0 will fail.

Reply:
> One last ask is we should add a test for this. You can use something like https://github.com/redis/redis/blob/unstable/tests/unit/type/list.tcl#L1276 for reference. I am imagining a simple test like:
> 
> ```
> 
> Create a deferring client.
> 
> Have it block with 0.001. 
> 
> Wait for the number of blocked clients to equal 0
> 
> Validate the empty read from the deferring client.
> 
> ```
> 
> 
> 
> This way, even if the client does block, the test won't hang since the wait for the number of blocked clients to equal 0 will fail. 

Sure thing, I'll do it on Sunday and update.

Reply:
Done, test is passing and without the fix - it's failing as expected.